service/gcs/core/gcs/uvm.go

```diff
@@ -96,6 +100,76 @@ func (h *Host) CreateContainer(id string, settings *prot.VMHostedContainerSettin
 		return c, nil
 	}
 
+	isSandboxOrStandalone := false
+	if criType, ok := settings.OCISpecification.Annotations["io.kubernetes.cri.container-type"]; ok {
+		// If the annotation is present it must be "sandbox"
+		isSandboxOrStandalone = criType == "sandbox"
+	} else {
+		// No annotation declares a standalone LCOW
+		isSandboxOrStandalone = true
+	}
+
+	// Check if we have a network namespace and if so generate the resolv.conf
+	// so we can bindmount it into the container
+	var networkNamespace string
+	if settings.OCISpecification.Windows != nil &&
+		settings.OCISpecification.Windows.Network != nil &&
+		settings.OCISpecification.Windows.Network.NetworkNamespace != "" {
+		networkNamespace = strings.ToLower(settings.OCISpecification.Windows.Network.NetworkNamespace)
+	}
+	if networkNamespace != "" {
+		if isSandboxOrStandalone {
+			// We have a network namespace. Generate the resolv.conf and add the bind mount.
+			netopts := h.netNamespaces[networkNamespace]
+			if netopts == nil || len(netopts.Adapters) == 0 {
+				logrus.WithFields(logrus.Fields{
+					"cid":   id,
+					"netNS": networkNamespace,
+				}).Warn("opengcs::CreateContainer - No adapters in namespace for sandbox")
+			} else {
+				logrus.WithFields(logrus.Fields{
+					"cid":   id,
+					"netNS": networkNamespace,
+				}).Info("opengcs::CreateContainer - Generate sandbox resolv.conf")
+
+				// TODO: Can we ever have more than one nic here at start?
+				td, err := ioutil.TempDir("", "")
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to create tmp directory")
+				}
+				resolvPath := filepath.Join(td, "resolv.conf")
+				adp := netopts.Adapters[0]
+				err = generateResolvConfFile(h.osl, resolvPath, prot.NetworkAdapter{
+					AdapterInstanceID: adp.ID,
+					HostDNSServerList: adp.DNSServerList,
+					HostDNSSuffix:     adp.DNSSuffix,
+				})
+				if err != nil {
+					return nil, err
+				}
+				// Store the resolve path for all workload containers
+				netopts.ResolvPath = resolvPath
+			}
+		}
+		if netopts, ok := h.netNamespaces[networkNamespace]; ok && netopts.ResolvPath != "" {
+			logrus.WithFields(logrus.Fields{
+				"cid":        id,
+				"netNS":      networkNamespace,
+				"resolvPath": netopts.ResolvPath,
+			}).Info("opengcs::CreateContainer - Adding /etc/resolv.conf mount")
+
+			settings.OCISpecification.Mounts = append(settings.OCISpecification.Mounts, oci.Mount{
+				Destination: "/etc/resolv.conf",
+				Type:        "bind",
+				Source:      netopts.ResolvPath,
+				Options:     []string{"bind", "ro"},
+			})
+		}
+	}
+
+	// Clear the windows section of the config
+	settings.OCISpecification.Windows = nil

```

Comment:
Is this just to indicate that you've done the conversion? And not strictly necessary?

Reply:
Yea. I mostly just want to remove the section as we really should not be passing it to `runc create` as its not a valid Linux config. But they dont seem to crash if we do. Just seems safer to remove.